include/mc_tasks/SplineTrajectoryTask.hpp

```diff
@@ -93,14 +96,76 @@ std::function<bool(const mc_tasks::MetaTask &, std::string &)> SplineTrajectoryT
 }
 
 template<typename Derived>
-void SplineTrajectoryTask<Derived>::update(mc_solver::QPSolver &)
+void SplineTrajectoryTask<Derived>::load(mc_solver::QPSolver & solver, const mc_rtc::Configuration & config)
+{
+  TrajectoryBase::load(solver, config);
+  if(config.has("gainsInterpolation"))
+  {
+    /**
+     * Load gains from config as vector<pair<double,Vector6d>> or vector<pair<double, double>>
+     **/
+    auto genValues = [this](const mc_rtc::Configuration & c, const std::string & key) {
+      auto config = c(key);
+      auto values = SequentialInterpolator6d::TimedValueVector{};
+      try
+      {
+        values = config;
+      }
+      catch(mc_rtc::Configuration::Exception & e)
+      {
+        e.silence();
+        try
+        {
+          std::vector<std::pair<double, double>> val = config;
+          values.reserve(val.size());
+          for(const auto & v : val)
+          {
+            values.push_back({v.first, v.second * Eigen::Vector6d::Ones()});
+          }
+        }
+        catch(mc_rtc::Configuration::Exception & e)
+        {
+          e.silence();
+          auto msg = fmt::format("[{}] {} interpolation values are expected to be either vector<pair<double, "
+                                 "Vector6d>> or vector<pair<double, double>>",
+                                 name(), key);
+          mc_rtc::log::critical(msg);
+          throw mc_rtc::Configuration::Exception(msg, config);
+        }
+      }
+      return values;
+    };
```

Comment:
I think it's better to allow the user to specify double/Vector6d at every time point

So something like:
```c++
auto genValues = [this](const mc_rtc::Configuration & c, const std::string & key) {
  std::vector<std::pair<double, mc_rtc::Configuration>> values = c;
  std::vector<std::pair<double, Eigen::Vector6d>> out;
  out.reserve(values.size());
  for(const auto & v : values)
  {
    if(v.second.size())
    {
      out.push_back({v.first, static_cast<Eigen::Vector6d>(v.second)});
    }
    else
    {
      out.push_back({v.first, Eigen::Vector6d::Constant(static_cast<double>(v.second))});
    }
  }
  return out;
};

Reply:
Good idea!
Thanks for the code snippet :+1: